lib/ansible/plugins/filter/core.py

```diff
@@ -283,11 +283,19 @@ def mandatory(a):
 
 def combine(*terms, **kwargs):
     recursive = kwargs.get('recursive', False)
-    if len(kwargs) > 1 or (len(kwargs) == 1 and 'recursive' not in kwargs):
-        raise AnsibleFilterError("'recursive' is the only valid keyword argument")
+    reverse = kwargs.get('reverse', False)
+
+    arg_error = "'recursive' and 'reverse' is the only valid keyword arguments"
+
+    for key in kwargs.keys():
```

Comment:
You can also do `for key in kwargs:`. By default, Python will only iterate over the keys; you have to explicitly specify `.items()` to iterate over `key, value` pairs.

Reply:
Ah, indeed.